core/core-auto-completion.el

```diff
@@ -34,10 +34,11 @@ The initialization function is hooked to `MODE-hook'."
               'company)
          (set (make-variable-buffer-local 'company-backends)
               ,backend-list))
-       (when auto-completion-enable-snippets-in-popup
-         (setq ,backend-list (mapcar 'spacemacs//show-snippets-in-company
-                                     ,backend-list)))
        (add-hook ',mode-hook ',func t)
+       (add-hook ',mode-hook (lambda ()
+                               (when auto-completion-enable-snippets-in-popup
+                                 (setq ,backend-list (mapcar 'spacemacs//show-snippets-in-company
+                                                             ,backend-list)))))
```

Comment:
It actually doesn't work if I add it to that function, only in the lambda for some reason. I've added a comment explaining why the lambda.